llvm/test/TableGen/x86-fold-tables.inc

```diff
@@ -5995,3 +5995,2634 @@ static const X86FoldTableEntry Table4[] = {
+  {X86::VCVTTPS2UDQZ128rrkz, X86::VCVTTPS2UDQZ128rmbkz, TB_BCAST_SS},
+  {X86::VCVTTPS2UDQZ256rrkz, X86::VCVTTPS2UDQZ256rmbkz, TB_BCAST_SS},
+  {X86::VCVTTPS2UDQZrrkz, X86::VCVTTPS2UDQZrmbkz, TB_BCAST_SS},
+  {X86::VCVTTPS2UQQZ128rrkz, X86::VCVTTPS2UQQZ128rmbkz, TB_BCAST_SS},
+  {X86::VCVTTPS2UQQZ256rrkz, X86::VCVTTPS2UQQZ256rmbkz, TB_BCAST_SS},
+  {X86::VCVTTPS2UQQZrrkz, X86::VCVTTPS2UQQZrmbkz, TB_BCAST_SS},
+  {X86::VCVTUDQ2PDZ128rrkz, X86::VCVTUDQ2PDZ128rmbkz, TB_BCAST_SD},
+  {X86::VCVTUDQ2PDZ256rrkz, X86::VCVTUDQ2PDZ256rmbkz, TB_BCAST_SD},
+  {X86::VCVTUDQ2PDZrrkz, X86::VCVTUDQ2PDZrmbkz, TB_BCAST_SD},
+  {X86::VCVTUDQ2PHZ128rrkz, X86::VCVTUDQ2PHZ128rmbkz, TB_BCAST_SH},
+  {X86::VCVTUDQ2PHZ256rrkz, X86::VCVTUDQ2PHZ256rmbkz, TB_BCAST_SH},
+  {X86::VCVTUDQ2PHZrrkz, X86::VCVTUDQ2PHZrmbkz, TB_BCAST_SH},
+  {X86::VCVTUDQ2PSZ128rrkz, X86::VCVTUDQ2PSZ128rmbkz, TB_BCAST_SS},
+  {X86::VCVTUDQ2PSZ256rrkz, X86::VCVTUDQ2PSZ256rmbkz, TB_BCAST_SS},
+  {X86::VCVTUDQ2PSZrrkz, X86::VCVTUDQ2PSZrmbkz, TB_BCAST_SS},
+  {X86::VCVTUQQ2PDZ128rrkz, X86::VCVTUQQ2PDZ128rmbkz, TB_BCAST_SD},
+  {X86::VCVTUQQ2PDZ256rrkz, X86::VCVTUQQ2PDZ256rmbkz, TB_BCAST_SD},
+  {X86::VCVTUQQ2PDZrrkz, X86::VCVTUQQ2PDZrmbkz, TB_BCAST_SD},
+  {X86::VCVTUQQ2PHZ128rrkz, X86::VCVTUQQ2PHZ128rmbkz, TB_BCAST_SH},
+  {X86::VCVTUQQ2PHZ256rrkz, X86::VCVTUQQ2PHZ256rmbkz, TB_BCAST_SH},
+  {X86::VCVTUQQ2PHZrrkz, X86::VCVTUQQ2PHZrmbkz, TB_BCAST_SH},
+  {X86::VCVTUQQ2PSZ128rrkz, X86::VCVTUQQ2PSZ128rmbkz, TB_BCAST_SS},
+  {X86::VCVTUQQ2PSZ256rrkz, X86::VCVTUQQ2PSZ256rmbkz, TB_BCAST_SS},
+  {X86::VCVTUQQ2PSZrrkz, X86::VCVTUQQ2PSZrmbkz, TB_BCAST_SS},
+  {X86::VCVTUW2PHZ128rrkz, X86::VCVTUW2PHZ128rmbkz, TB_BCAST_SH},
+  {X86::VCVTUW2PHZ256rrkz, X86::VCVTUW2PHZ256rmbkz, TB_BCAST_SH},
+  {X86::VCVTUW2PHZrrkz, X86::VCVTUW2PHZrmbkz, TB_BCAST_SH},
+  {X86::VCVTW2PHZ128rrkz, X86::VCVTW2PHZ128rmbkz, TB_BCAST_SH},
+  {X86::VCVTW2PHZ256rrkz, X86::VCVTW2PHZ256rmbkz, TB_BCAST_SH},
+  {X86::VCVTW2PHZrrkz, X86::VCVTW2PHZrmbkz, TB_BCAST_SH},
+  {X86::VDIVPDZ128rr, X86::VDIVPDZ128rmb, TB_BCAST_SD},
+  {X86::VDIVPDZ256rr, X86::VDIVPDZ256rmb, TB_BCAST_SD},
+  {X86::VDIVPDZrr, X86::VDIVPDZrmb, TB_BCAST_SD},
+  {X86::VDIVPHZ128rr, X86::VDIVPHZ128rmb, TB_BCAST_SH},
+  {X86::VDIVPHZ256rr, X86::VDIVPHZ256rmb, TB_BCAST_SH},
+  {X86::VDIVPHZrr, X86::VDIVPHZrmb, TB_BCAST_SH},
+  {X86::VDIVPSZ128rr, X86::VDIVPSZ128rmb, TB_BCAST_SS},
+  {X86::VDIVPSZ256rr, X86::VDIVPSZ256rmb, TB_BCAST_SS},
+  {X86::VDIVPSZrr, X86::VDIVPSZrmb, TB_BCAST_SS},
+  {X86::VEXP2PDZrkz, X86::VEXP2PDZmbkz, TB_BCAST_SD},
+  {X86::VEXP2PSZrkz, X86::VEXP2PSZmbkz, TB_BCAST_SS},
+  {X86::VFCMULCPHZ128rr, X86::VFCMULCPHZ128rmb, TB_BCAST_SH},
+  {X86::VFCMULCPHZ256rr, X86::VFCMULCPHZ256rmb, TB_BCAST_SH},
+  {X86::VFCMULCPHZrr, X86::VFCMULCPHZrmb, TB_BCAST_SH},
+  {X86::VFMULCPHZ128rr, X86::VFMULCPHZ128rmb, TB_BCAST_SH},
+  {X86::VFMULCPHZ256rr, X86::VFMULCPHZ256rmb, TB_BCAST_SH},
+  {X86::VFMULCPHZrr, X86::VFMULCPHZrmb, TB_BCAST_SH},
```

Comment:
The FMULCPH/FCMULCPH fp16 complex instructions are 32-bit broadcasts (2 x fp16) - not sure if there are others?

Reply:
I think we can safely perform this with a custom entry using TB_BCAST_SS, but it would be OK just to disable them for now (and add a TODO comment).

Reply:
Done https://github.com/llvm/llvm-project/pull/73654/commits/56fd306e83cfad6f1495549eabf3b4a773f64186
It's easy so I fixed it directly.